File: C1.UWP.OrgChart/CS/OrgChartSamples/Person.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace OrgChartSamples
{
    /// <summary>
    /// Our hierarchical data item: A Person has Subordinates of type Person.
    /// </summary>
    public class Person
    {
        ObservableCollection<Person> _list = new ObservableCollection<Person>();

        #region ** object model

        public string Name { get; set; }
        public string Position { get; set; }
        public string Notes { get; set; }
        public IList<Person> Subordinates
        {
            get { return _list; }
        }
        public int TotalCount
        {
            get
            {
                var count = 1;
                foreach (var p in Subordinates)
                {
                    count += p.TotalCount;
                }
                return count;
            }
        }
        public override string ToString()
        {
            return string.Format("{0}:\r\n\t{1}", Name, Position);
        }

        #endregion

        #region ** Person factory

        static Random _rnd = new Random();
        static string[] _positions = (Strings.Director + "|" + Strings.Manager + "|" + Strings.Designer + "|" + Strings.StringPositions).Split('|');
        static string[] _areas = Strings.StringAreas.Split('|');
        static string[] _first = Strings.StringFirstName.Split('|');
        static string[] _last = Strings.StringLastName.Split('|');
        static string[] _verb = Strings.StringVerb.Split('|');
        static string[] _adjective = Strings.StringAdjective.Split('|');
        static string[] _noun = Strings.StringNoun.Split('|');

        public static Person CreatePerson(int level)
        {
            var p = CreatePerson();
            if (level > 0)
            {
                level--;
                if (Windows.Foundation.Metadata.ApiInformation.IsTypePresent("Windows.Phone.UI.Input.HardwareButtons"))
                {
                    for (int i = 0; i < _rnd.Next(3, 3); i++)
                    {
                        p.Subordinates.Add(CreatePerson(_rnd.Next(level / 2, level)));
                    }
                }
                else
                {
                    for (int i = 0; i < _rnd.Next(1, 4); i++)
                    {
                        p.Subordinates.Add(CreatePerson(_rnd.Next(level / 2, level)));
                    }
                }
                   
            }
            return p;
        }
        public static Person CreatePerson()
        {
            var p = new Person();
            if (Windows.Foundation.Metadata.ApiInformation.IsTypePresent("Windows.Phone.UI.Input.HardwareButtons"))
            {
                p.Position = string.Format("{0}", GetItem(_positions));
            }
            else
            {
                p.Position = string.Format(Strings.StringFormatTwoArg, GetItem(_positions), GetItem(_areas));
            }
            p.Name = string.Format("{0} {1}", GetItem(_first), GetItem(_last));
            p.Notes = string.Format("{0} {1} {2} {3}", p.Name, GetItem(_verb), GetItem(_adjective), GetItem(_noun));
            while (_rnd.NextDouble() < .5)
            {
                p.Notes += string.Format(Strings.StringFormatThreeArg, GetItem(_verb), GetItem(_adjective), GetItem(_noun));
            }
            p.Notes += ".";
            return p;
        }
        static string GetItem(string[] list)
        {
            return list[_rnd.Next(0, list.Length)];
        }

        #endregion
    }
}

File: C1.UWP.OrgChart/VB/OrgChartSamples/Person.vb
Imports System.Collections.ObjectModel
Imports System.Collections.Generic
Imports System.Collections
Imports System

''' <summary>
''' Our hierarchical data item: A Person has Subordinates of type Person.
''' </summary>
Public Class Person
    Private _list As New ObservableCollection(Of Person)()

#Region "** object model"

    Public Property Name() As String
    Public Property Position() As String
    Public Property Notes() As String
    Public ReadOnly Property Subordinates() As IList(Of Person)
        Get
            Return _list
        End Get
    End Property
    Public ReadOnly Property TotalCount() As Integer
        Get
            Dim count As Integer = 1
            For Each p As Person In Subordinates
                count += p.TotalCount
            Next
            Return count
        End Get
    End Property
    Public Overrides Function ToString() As String
        Return String.Format("{0}:" & vbCr & vbLf & vbTab & "{1}", Name, Position)
    End Function

#End Region

#Region "** Person factory"

    Shared _rnd As New Random()
    Shared _positions As String() = (Strings.Director + "|" + Strings.Manager + "|" + Strings.Designer + "|" + Strings.StringPositions).Split("|"c)
    Shared _areas As String() = Strings.StringAreas.Split("|"c)
    Shared _first As String() = Strings.StringFirstName.Split("|"c)
    Shared _last As String() = Strings.StringLastName.Split("|"c)
    Shared _verb As String() = Strings.StringVerb.Split("|"c)
    Shared _adjective As String() = Strings.StringAdjective.Split("|"c)
    Shared _noun As String() = Strings.StringNoun.Split("|"c)

    Public Shared Function CreatePerson(level As Integer) As Person
        Dim p As Person = CreatePerson()
        If level > 0 Then
            level -= 1
            If Metadata.ApiInformation.IsTypePresent("Windows.Phone.UI.Input.HardwareButtons") Then
                Dim i As Integer = 0
                While i < _rnd.Next(3, 3)
                    Dim halfLevel As Double = level / 2
                    p.Subordinates.Add(CreatePerson(_rnd.Next(CInt(level / 2), level)))
                    i += 1
                End While
            Else
                Dim i As Integer = 0
                While i < _rnd.Next(1, 4)
                    p.Subordinates.Add(CreatePerson(_rnd.Next(CInt(level / 2), level)))
                    i += 1
                End While

            End If
        End If
        Return p
    End Function
    Public Shared Function CreatePerson() As Person
        Dim p As Person = New Person()
        If Metadata.ApiInformation.IsTypePresent("Windows.Phone.UI.Input.HardwareButtons") Then
            p.Position = String.Format("{0}", GetItem(_positions))
        Else
            p.Position = String.Format(Strings.StringFormatTwoArg, GetItem(_positions), GetItem(_areas))
        End If
        p.Name = String.Format("{0} {1}", GetItem(_first), GetItem(_last))
        p.Notes = String.Format("{0} {1} {2} {3}", p.Name, GetItem(_verb), GetItem(_adjective), GetItem(_noun))
        While _rnd.NextDouble() < 0.5
            p.Notes += String.Format(Strings.StringFormatThreeArg, GetItem(_verb), GetItem(_adjective), GetItem(_noun))
        End While
        p.Notes += "."
        Return p
    End Function
    Shared Function GetItem(list As String()) As String
        Return list(_rnd.[Next](0, list.Length))
    End Function

#End Region
End Class